platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/configurations/DefaultConfiguration.java

```diff
@@ -662,7 +663,7 @@ private class ConfigurationResolutionAccess implements ResolutionAccess {
 
         @Override
         public ResolutionHost getHost() {
-            return new DefaultResolutionHost(DefaultConfiguration.this);
+            return new DefaultResolutionHost(DefaultConfiguration.this, DefaultConfiguration.this.defaultConfigurationFactory.getProblems());
```

Comment:
This is the reason for injecting the `Problems` service into the `DefaultConfigurationFactory` - we need to have it available when creating `ResolutionHost` implementations.  Could also inject into `DefaultConfiguration` directly, but that seemed like it would be more of a pain.

Reply:
As long as we have the factory, I'm not terribly concerned with how this works.  I just thought it might look weird, so was pointing out why I did it like this.